prow/tide/tide.go

```diff
@@ -246,8 +253,15 @@ func toSimpleState(s kube.ProwJobState) simpleState {
 
 // isPassingTests returns whether or not all contexts set on the PR except for
 // the tide pool context are passing.
-func isPassingTests(pr PullRequest) bool {
-	for _, ctx := range pr.Commits.Nodes[0].Commit.Status.Contexts {
+func isPassingTests(log *logrus.Entry, ghc githubClient, pr PullRequest) bool {
```

Comment:
Consider returning `(bool, error)`.

Reply:
I had it that way originally, but since the error handling is always the same I moved it inside the function. If we can't determine if the tests are passing we should always log an error and then proceed as if the tests aren't passing.
I can change this if the `(bool, error)` return type is preferred, that does seem more idiomatic to me.

Reply:
Either way is fine with me. I liked [this article](https://blog.merovius.de/2018/01/21/what_even_is_error_handling.html) about go error handling.